src/pragma.js

```diff
@@ -0,0 +1,66 @@
+"use strict";
+
+const parse = require("./parser");
+const { EOL } = require("os");
+
+const extractDocBlocks = text => {
+  const parsed = parse(text);
```

Comment:
any idea how often this gets called or if there's any performance concern? (I'm guessing once per file which probably isn't a huge deal)

Reply:
Yup, just checked:

- once per file, if you use no flags or `--require-pragma`
- twice per file, if you use `--insert-pragma` (or both flags)

Some room for optimization.  I didn't know that this hook is still called even when you use no flags.

Reply:
I'll do a couple quick optimizations here, give me a bit.

Reply:
Test suite slowed down by ~1% with my original patch, now closer to .1%.  Pretty much noise.

LMK if you don't want the new npm dependency, memoize impls are pretty interchangeable.